src/odemis/util/test/transform_test.py

```diff
@@ -334,10 +334,16 @@ class PixelIndexCoordinateTransformBase(unittest.TestCase):
 
     def setUp(self):
         self._ji = [(0, 0), (0, 4), (7, 0), (7, 4), (0.5, 0.5)]
-        self._xy = [(-2, 3.5), (2, 3.5), (-2, -3.5), (2, -3.5), (-1.5, 3)]
-        self._xy2 = [(-4, 7), (4, 7), (-4, -7), (4, -7), (-3, 6)]
-        self._xy23 = [(-4, 10.5), (4, 10.5), (-4, -10.5), (4, -10.5), (-3, 9)]
-        self._shape = (8, 5)
+
+        # (shape, pixel_size, xy)
+        self._known_values = [
+            ((8, 5), None, [(-2, 3.5), (2, 3.5), (-2, -3.5), (2, -3.5), (-1.5, 3)]),
+            ((8, 5), 2, [(-4, 7), (4, 7), (-4, -7), (4, -7), (-3, 6)]),
+            ((8, 5), (2, 3), [(-4, 10.5), (4, 10.5), (-4, -10.5), (4, -10.5), (-3, 9)]),
+            (None, None, [(0, 0), (4, 0), (0, -7), (4, -7), (0.5, -0.5)]),
+            (None, 2, [(0, 0), (8, 0), (0, -14), (8, -14), (1, -1)]),
+            (None, (2, 3), [(0, 0), (8, 0), (0, -21), (8, -21), (1, -1.5)]),
+        ]
```

Comment:
This long list of variables is a confusing, it is hard to see what these list mean and how the numbers are organized.
Maybe instead you could use a dict, like:
> self._shape_test_values = {(0, 0):(8, 5) , (0, 4):(8, 5) , (7, 0):(8, 5) etc. }
> self._pixel_size_test_values = {(0, 0):None , (0, 4):2 , (7, 0):(2, 3) etc.}
> self._xy_test_values = {
(0, 0):[(-2, 3.5), (2, 3.5), (-2, -3.5), (2, -3.5), (-1.5, 3)],
(0, 4):[(-4, 7), (4, 7), (-4, -7), (4, -7), (-3, 6)],
(7, 0):[(-4, 10.5), (4, 10.5), (-4, -10.5), (4, -10.5), (-3, 9)]
etc. }

Or otherwise three separate lists for each column one.
Maybe this could also  solve the double loop problem: https://github.com/delmic/odemis/pull/1575/files#r527004842 (using a combination of itertools and zip)

Reply:
Changed structure and added comment to clarify